### PR TITLE
Errors have optional `jsonUrl` and `jsonPosition` fields.

### PR DESCRIPTION
**Pull request by sergey-shandar**

### CODE REVIEW

File: lib/util/errorFileInfo.ts [hunk 1]
```diff
@@ -25,6 +33,29 @@ const errorAddFileInfo = <T extends NodeError<T>>(error: T): void => {
...
+  /*
+  const schema = errorEx[schemaSymbol]
+  if (schema !== null && typeof schema === "object") {
+    const schemaInfo = getInfo(schema)
+    if (schemaInfo !== undefined) {
+      error.position = schemaInfo.position
+      error.url = getRootObjectInfo(schemaInfo).url
+    }
+  }
+  */
```
Comment (vladbarosan) [inline lib/util/errorFileInfo.ts]: commented code ?

File: lib/util/errorFileInfo.ts [hunk 2]
```diff
@@ -25,6 +33,29 @@ const errorAddFileInfo = <T extends NodeError<T>>(error: T): void => {
       log.error(`ICE: can't parse title: ${title}`)
     }
   }
+  const errorEx = error as any as ErrorEx
```
Comment (vladbarosan) [inline lib/util/errorFileInfo.ts]: > error as any as ErrorEx [](start = 17, length = 24)

ehh

File: lib/util/errorFileInfo.ts (hunk 1 above)
Comment (sergey-shandar) [reply lib/util/errorFileInfo.ts]: let me remove it

File: lib/util/errorFileInfo.ts [hunk 3]
```diff
@@ -25,6 +33,29 @@ const errorAddFileInfo = <T extends NodeError<T>>(error: T): void => {
...
+  const json = errorEx[jsonSymbol]
+  if (json !== undefined) {
+    const jsonInfo = getInfo(json)
+    if (jsonInfo !== undefined) {
+      const path = error.path
+      error.jsonPosition = getDescendantFilePosition(
+        json,
+        path === undefined ? undefined : isArray(path) ? path : path.split("/")
+      )
+      error.jsonUrl = getRootObjectInfo(jsonInfo).url
```
Comment (vladbarosan) [inline lib/util/errorFileInfo.ts]: >getRootObjectInfo(jsonInfo).url [](start = 22, length = 31)

is this url always going to be there or do we need to check ?

File: lib/util/errorFileInfo.ts [hunk 4]
```diff
@@ -25,6 +33,29 @@ const errorAddFileInfo = <T extends NodeError<T>>(error: T): void => {
       log.error(`ICE: can't parse title: ${title}`)
     }
   }
+  const errorEx = error as any as ErrorEx
+  const json = errorEx[jsonSymbol]
+  if (json !== undefined) {
+    const jsonInfo = getInfo(json)
```
Comment (vladbarosan) [inline lib/util/errorFileInfo.ts]: Maybe a test for this

File: lib/util/errorFileInfo.ts (hunk 3 above)
Comment (sergey-shandar) [reply lib/util/errorFileInfo.ts]: It's optional so it could be `undefined`.

**Pull request merged**